feat(curriculum): add binary search tree (#52371)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/20-upcoming-python/learn-tree-traversal-by-building-a-binary-search-tree/655a1564f3aa8210938cdf68.md

---
id: 655a1564f3aa8210938cdf68
title: Step 17
challengeType: 20
dashedName: step-17
---

# --description--

> Add another condition using `elif` that checks if the `key` > `node.key`.

Reviewer: ```suggestion
Add an `elif` conditional statement that checks if `key` > `node.key`.
```

# --hints--

You should use the `elif` keyword for the second condition.

```js
({ test: () => assert.match(code, /elif/) });
```

Check if the `key` > `node.key`.

```js
({ test: () => assert.match(code, /key\s*>\s*node\.key/) });

> ```

Reviewer: As for the previous step I would replace these two with:

You should add an `elif` conditional statement to check if `key > node.key`.

```js
({ test: () => assert.match(e.code.original["main.py"], /^(\s+)if\s+node\s+is\s+None\s*:.*?\1elif\s+key\s*>\s*node\.key\s*:/ms) })
```

# --seed--

## --seed-contents--

```py
class TreeNode:
    def __init__(self, key):
        self.key = key
        self.left = None
        self.right = None

class BinarySearchTree:
    def __init__(self):
        self.root = None

    def insert(self,key):
        self.root = self._insert(self.root, key)

    def _insert(self, node, key):
        if node is None:
            return TreeNode(key)

>         if key < node.key:
>             node.left = self._insert(node.left, key)
> --fcc-editable-region--
>             
> --fcc-editable-region--

Reviewer: ```suggestion
--fcc-editable-region--
        if key < node.key:
            node.left = self._insert(node.left, key)            
--fcc-editable-region--
```
